core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -449,6 +449,13 @@ default ByteBufAllocator alloc() {
     @UnstableApi
     ExchangeType exchangeType();
 
+    /**
+     * Initiates connection shutdown and returns {@link CompletableFuture} that completes when the connection
+     * associated with this context is closed.
```

Comment:
How about adding some specifics about the underlying behavior? For example:

```
If the underlying connection's protocol is HTTP/1, {@code "connection: close"} header
is sent with the request (if not sent already) and the connection will be closed
as soon as all pending requests are processed. Otherwise, a GOAWAY frame will be sent
to initiate graceful connection shutdown.
```